filebeat/harvester/reader/json.go

```diff
@@ -120,6 +120,9 @@ func (r *JSON) Next() (Message, error) {
 	if err != nil {
 		return message, err
 	}
-	message.Content, message.Fields = r.decodeJSON(message.Content)
+
+	var fields = common.MapStr{}
+	message.Content, fields = r.decodeJSON(message.Content)
+	message.AddFields(common.MapStr{"json": fields})
```

Comment:
I think if fields is nil probably something went wrong. But we also didn't catch this previously and I don't intend to do "logic" changes in this PR. 

Probably decodeJSON should be able to return an error.